arch/riscv/core/isr.S

```diff
@@ -215,6 +215,20 @@
 	li temp, MSTATUS_MPP					;\
 	and ret, ret, temp					;
 
+#define GET_CPU(ret, temp) \
+	csrr ret, mhartid		;\
+	li temp, ___cpu_t_SIZEOF	;\
+	mul temp, temp, ret		;\
```

Comment:
@edersondisouza - mul is part of the M extension which is optional and might not present in small cpus. Additionally, I think it's very expensive to execute multiplications in ISRs that often (especially without SMP).

Do you think we should guard this line with an `ifdef CONFIG_SMP`? Pretty should all multi-core cpus have the M extensions and rather small single-core wouldn't.

Reply:
Oh, true. I guess that checking for `CONFIG_SMP` would make a quick fix. Do you have a patch in hand for it?

More generally, I'm not sure how expensive `mul` is compared to just looping and adding - another approach. I mean, if `mul` makes sense for some scenarios, maybe a full solution would entail some check for M extension being available.

Reply:
@edersondisouza - I will create a PR and tag you for reviewing this week. My idea is to check if M extension is present and otherwise use additions.

I checked the `mul` instruction again and it should be fine. Mixed it up with `div` which is really expensiv.